packages/ag-charts-community/src/util/object.ts

```diff
@@ -131,3 +131,22 @@ export function deepFreeze<T>(obj: T): T {
 
     return obj;
 }
+
+export function defineSetterGetter<T, P extends string, V>(
+    o: T,
+    p: P,
+    attributes: { set: (v: V) => void; get: () => V }
+): T & { [K in P]: V };
+export function defineSetterGetter<T, P extends string, V>(
+    o: T,
+    p: P,
+    attributes: { set?: undefined; get: () => V }
+): T & { readonly [K in P]: V };
+export function defineSetterGetter<T, P extends string, V>(
+    o: T,
+    p: P,
+    attributes: { set?: (v: V) => void; get: () => V }
+): T {
+    Object.defineProperty(o, p, attributes);
+    return o;
+}
```

Comment:
This keeps the same signature but without needing to disable linting rules:
```suggestion
export function defineSetterGetter<T, P extends string, V, S extends ((v: V) => void) | undefined>(
    o: T,
    p: P,
    attributes: { set?: S; get: () => V }
) {
    Object.defineProperty(o, p, attributes);
    return o as S extends undefined ? T & { readonly [K in P]: V } : T & { [K in P]: V };
}
```

Reply:
Thanks for the feedback. I've tried this out, but the return type isn't quite what you'd expect.
```typescript
const a = defineSetterGetter( {}, 'p', { get: () => 'foobar'} );
// a is of type:
// * `{ readonly p: 'foobar'} | { p: 'foobar' }` when using `as`
// * `{ readonly p: 'foobar'}`  when using an overload
```

The two types are the same in practice, but the overload is tidier.

Reply:
obsolete.